frontend/packages/console-dynamic-plugin-sdk/src/extensions/dashboards.ts

```diff
@@ -0,0 +1,383 @@
+  export type DashboardsOverviewHealthResourceSubsystem = {
+    /** The display name of the subsystem. */
+    title: string;
+    /** Kubernetes resources which will be fetched and passed to healthHandler */
+    resources: EncodedCodeRef;
+    /** Resolve the subsystem's health */
+    healthHandler: EncodedCodeRef;
+    /** Loader for popup content. If defined, a health item will be represented as a link which opens popup with given content. */
+    popupComponent?: EncodedCodeRef;
+    /** The title of the popover. */
+    popupTitle?: string;
+  };
+
+  export type DashboardsOverviewHealthResourceSubsystemCodeRefs<
+    T extends ResourcesObject = ResourcesObject
+  > = {
+    resources: CodeRef<WatchK8sResources<T>>;
+    healthHandler: CodeRef<ResourceHealthHandler<T>>;
+    popupComponent?: CodeRef<WatchK8sResults<T>>;
+  };
+
+  /** Adds a health subsystem to the status card of Overview dashboard where the source of status is a K8s REST API. */
+  export type DashboardsOverviewHealthOperator = {
+    /** Title of operators section in popup */
+    title: string;
+    /** Kubernetes resources which will be fetched and passed to healthHandler */
+    resources: EncodedCodeRef;
+    /** Resolves status for operators */
+    getOperatorsWithStatuses?: EncodedCodeRef;
+    /** Loader for popup row component */
+    operatorRowLoader?: EncodedCodeRef;
+    /** Links to all resources page. If not provided then a list page of the first resource from resources prop is used. */
+    viewAllLink?: string;
+  };
+
+  export type DashboardsOverviewHealthOperatorCodeRefs<
+    T extends K8sResourceCommon = K8sResourceCommon
+  > = {
+    resources: CodeRef<FirehoseResource[]>;
+    getOperatorsWithStatuses?: CodeRef<GetOperatorsWithStatuses<T>>;
+    operatorRowLoader?: React.ComponentType<OperatorRowProps<T>>;
+  };
+
+  /** Adds an inventory status group */
+  export type DashboardsInventoryItemGroup = {
+    /** The id of the status group. */
+    id: string;
+    /** React component representing status group icon. */
+    icon: EncodedCodeRef;
+  };
+
+  export type DashboardsInventoryItemGroupCodeRefs = {
+    icon: CodeRef<React.ReactElement>;
+  };
+
+  /** Adds a resource tile to the overview inventory card */
+  export type DashboardsOverviewInventoryItem = {
+    /** The model for `resource` which will be fetched. The model is used for getting model's label or abbr. */
+    model: EncodedCodeRef;
+    /** Function which will map various statuses to groups. */
+    mapper?: EncodedCodeRef;
+    /** Additional resources which will be fetched and passed to `mapper` function. */
+    additionalResources?: EncodedCodeRef;
+  };
+
+  export type DashboardsOverviewInventoryItemCodeRefs<
+    T extends K8sResourceCommon = K8sResourceCommon
+  > = {
+    model: CodeRef<K8sKind>;
+    mapper?: CodeRef<StatusGroupMapper>;
+    additionalResources?: CodeRef<WatchK8sResources<{ [key: string]: T | T[] }>>;
+  };
+
```

Comment:
This way we wouldnt be able to specify different types for different resources.
We should also add the generic for `StatusGroupMapper`
```suggestion
  export type DashboardsOverviewInventoryItemCodeRefs<
    T extends { [key: string]: K8sResourceCommon | K8sResourceCommon[] }
  > = {
    model: CodeRef<K8sKind>;
    mapper?: CodeRef<StatusGroupMapper<T>>;
    additionalResources?: CodeRef<WatchK8sResources<T>>;
  };

```